Unicode characters are not correctly parsed from byte[] if default charset is not UTF-8

The `TestUtf8.supportI18nBytes` test is currently broken on windows systems. Actually in my IDE (IntelliJ) the test is green. but the maven test target fails both in the IDE and on command line:

```
[ERROR] Failures:
[ERROR] TestUtf8.supportI18nBytes:62 Parsing bytes[] Amharic text ==> expected: <አማርኛ> but was: <áŠ áˆ›áˆ­áŠ›>
[ERROR] TestUtf8.supportI18nBytes:62 Parsing bytes[] Malayalam text ==> expected: <മലയാളം> but was: <à´®à´²à´¯à´¾à´³à´‚>
[ERROR] TestUtf8.supportI18nBytes:62 Parsing bytes[] Assyrian Neo-Aramaic text ==> expected: <ܐܬܘܪܝܐ> but was: <Ü?Ü¬Ü˜ÜªÜ?Ü?>
[ERROR] TestUtf8.supportI18nBytes:62 Parsing bytes[] Georgian text ==> expected: <მარგალური> but was: <áƒ›áƒ?áƒ áƒ’áƒ?áƒšáƒ£áƒ áƒ˜>
[ERROR] TestUtf8.supportI18nBytes:62 Parsing bytes[] Sinhala text ==> expected: <සිංහල ජාතිය> but was: <à·ƒà·’à¶‚à·„à¶½ à¶¢à·?à¶­à·’à¶º>
[ERROR] TestUtf8.supportI18nBytes:62 Parsing bytes[] Japanese text ==> expected: <日本語> but was: <æ—¥æœ¬èªž>
[ERROR] TestUtf8.supportI18nBytes:62 Parsing bytes[] Russian text ==> expected: <Русский> but was: <Ð ÑƒÑ?Ñ?ÐºÐ¸Ð¹>
[ERROR] TestUtf8.supportI18nBytes:62 Parsing bytes[] Farsi text ==> expected: <فارسی> but was: <Ù?Ø§Ø±Ø³ÛŒ>
[ERROR] TestUtf8.supportI18nBytes:62 Parsing bytes[] Korean text ==> expected: <한국어> but was: <í•œêµ­ì–´>
[ERROR] TestUtf8.supportI18nBytes:62 Parsing bytes[] Armenian text ==> expected: <Հայերեն> but was: <Õ€Õ¡ÕµÕ¥Ö€Õ¥Õ¶>
[ERROR] TestUtf8.supportI18nBytes:62 Parsing bytes[] Hindi text ==> expected: <हिन्दी> but was: <à¤¹à¤¿à¤¨à¥?à¤¦à¥€>
[ERROR] TestUtf8.supportI18nBytes:62 Parsing bytes[] Hebrew text ==> expected: <עברית> but was: <×¢×‘×¨×™×ª>
[ERROR] TestUtf8.supportI18nBytes:62 Parsing bytes[] Chinese text ==> expected: <中文> but was: <ä¸­æ–‡>
```

The problem seems to be the creation of `String` using the system's default charset in `JSONParserByteArray.extractString`

https://github.com/netplex/json-smart-v2/blob/604281d6b085f02b745c1ef8f04b056035acf127/json-smart/src/main/java/net/minidev/json/parser/JSONParserByteArray.java#L62

and `JSONParserByteArray.extractStringTrim`

https://github.com/netplex/json-smart-v2/blob/604281d6b085f02b745c1ef8f04b056035acf127/json-smart/src/main/java/net/minidev/json/parser/JSONParserByteArray.java#L74

--- a/json-smart/src/main/java/net/minidev/json/parser/JSONParserByteArray.java
+++ b/json-smart/src/main/java/net/minidev/json/parser/JSONParserByteArray.java
@@ -19,6 +19,9 @@ import static net.minidev.json.parser.ParseException.ERROR_UNEXPECTED_EOF;
 import net.minidev.json.JSONValue;
 import net.minidev.json.writer.JsonReaderI;
 
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+
 /**
  * Parser for JSON text. Please note that JSONParser is NOT thread-safe.
  * 
@@ -59,7 +62,7 @@ class JSONParserByteArray extends JSONParserMemory {
 	}
 
 	protected void extractString(int beginIndex, int endIndex) {
-		xs = new String(in, beginIndex, endIndex - beginIndex);
+		xs = new String(in, beginIndex, endIndex - beginIndex, StandardCharsets.UTF_8);
 	}
 
 	protected void extractStringTrim(int start, int stop) {
@@ -71,7 +74,7 @@ class JSONParserByteArray extends JSONParserMemory {
 		while ((start < stop) && (val[stop - 1] <= ' ')) {
 			stop--;
 		}
-		xs = new String(in, start, stop - start);
+		xs = new String(in, start, stop - start, StandardCharsets.UTF_8);
 	}
 
 	protected int indexOf(char c, int pos) {

--- a/json-smart/src/test/java/net/minidev/json/test/TestUtf8.java
+++ b/json-smart/src/test/java/net/minidev/json/test/TestUtf8.java
@@ -19,7 +19,8 @@ public class TestUtf8 {
 				Arguments.of("Russian", "Русский"), Arguments.of("Farsi", "فارسی"), Arguments.of("Korean", "한국어"),
 				Arguments.of("Armenian", "Հայերեն"), Arguments.of("Hindi", "हिन्दी"), Arguments.of("Hebrew", "עברית"),
 				Arguments.of("Chinese", "中文"), Arguments.of("Amharic", "አማርኛ"), Arguments.of("Malayalam", "മലയാളം"),
-				Arguments.of("Assyrian Neo-Aramaic", "ܐܬܘܪܝܐ"), Arguments.of("Georgian", "მარგალური"));
+				Arguments.of("Assyrian Neo-Aramaic", "ܐܬܘܪܝܐ"), Arguments.of("Georgian", "მარგალური"),
+				Arguments.of("Emojis", "🐶🐱🐭🐹🐰🦊🐻🐼🐻‍❄🐨🐯🦁🐮🐷🐽🐸🐵🙈🙉🙊🐒🐔🐧🐦🐤🐣🐥🦆🦅🦉🦇🐺🐗🐴🦄🐝🐛"));
 	};
 
 	@ParameterizedTest

